src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -435,39 +435,40 @@ public static char binaryBeMsb0ToHexDigit(final boolean[] src) {
      * @return a hexadecimal digit representing the selected bits
      * @throws IllegalArgumentException if {@code src} is empty
      * @throws NullPointerException if {@code src} is {@code null}
+     * @throws IndexOutOfBoundsException if {@code srcPos} is outside the array.
      */
     public static char binaryBeMsb0ToHexDigit(boolean[] src, int srcPos) {
         if (src.length == 0) {
```

Comment:
To leave only one branch check on the main path I would put this check inside the other check:
```Java
if (Integer.compareUnsigned(srcPos, src.length) >= 0) {
    // Throw the correct exception
    if (src.length == 0) {
        throw new IllegalArgumentException("Cannot convert an empty array.");
    }
    throw new IndexOutOfBoundsException(srcPos + " is not within array length " + src.length);
}
```